docs/approach-docs/platform-containers.md

```diff
@@ -0,0 +1,98 @@
+
+## Details
+
+Installing postgres directly on a development machine might not be possible for some, and requires a level of support and maintenance on their part. With a containerized approach, we can eliminate this hurdle and perhaps shorten the time it takes for new developers to get setup. 
+
+Additionally, we could have a more consistent database environment for all developers, for instance, setting up users and other related artifacts during the containerization process.
+
+When I first setup Journaly, I used docker for my DB, and with the following changes, had it up and running in little time.
+
+### Docker
+
+We need a root level `Dockerfile` that looks something like this:
+
+```docker
+FROM postgres:11
+
+EXPOSE 5432
+
+COPY bootstrap.sh /docker-entrypoint-initdb.d/db-bootstrap.sh
+```
+
+You'll see that I have a bootstrap shell script to set up some initial database work:
+
+```sh
+#!/bin/bash
+
+set -e # bail immediately on error
+
+psql <<- EOSQL
+    CREATE USER developer WITH CREATEDB PASSWORD 'nights_watch';
+    CREATE DATABASE journaly_db;
+EOSQL
+
+echo "Database bootstrap complete."
+```
+
+There's a lot of potential here to automate environment and database settings.
+
+### Docker-Compose
+
+Finally, we need a `docker-compose.yml` file to tie it all together:
+
+```yaml
+version: '3'
+
+services:
+  journaly:
+    image: 'journaly:latest'
+    ports:
+      - '5432:5432'
+    volumes:
+      - journaly_db:/var/lib/postgresql/data
+    environment:
+      POSTGRES_PASSWORD: <db_password>
+    build: .
+
+volumes:
+  journaly_db:
+
+```
+
+### Final Modifications
+
+With the given files, we end up with the following additions:
+
+- `db-bootstrap.sh`
+- `Dockerfile`
+- `docker-compose.yaml`
+
+We can then use compose with the following commands to build and run the image and container:
+
+```bash
+$ docker-compose build
+$ docker-compose up
+```
+
+Nothing else changes for the setup instructions, the database is still exposed on the default postgres port `5432` and the database seed can run directly against the container.
+
+## Concerns
+
+### Docker Knowledge
+
+I found docker difficult to grasp when it came out and still struggle with advanced concepts. We'd be asking our developers to have some level of understanding of docker to work with our stack. If they don't have docker at all, it would require an install of additional software.
+
+At this point, it would be optional, and an alternative to installing postgres.
+
+### Permissions
+
+You'll notice that the `docker-compose.yml` file has a line for the postgres setting `POSTGRES_PASSWORD`. This is required for postgres to run, which means yet another password for the developer to setup. The alternative is to set postgres to run completely insecure, not a great option even though it's for local development.
```

Comment:
I'm OK with this being insecure since we probably won't be productionalizing the docker env. We can just hardcode a password. If we dockerize the app as well, I don't think we need to `EXPOSE` postgress outside of the docker compose sphere so it won't really matter if the password is common knowledge, no one outside of the dockerized env will be able to connect to PG.

Reply:
The only issue I see is if a developer wants to query or update local dev data with a client, they won't be able to connect to it. Also, npm scripts for database updates wouldn't be able to connect.

Reply:
The workflow I'm used to is that you `docker exec` into a container to run stuff like NPM scripts or a DB shell, although I suppose it is kind of an annoying extra step.

I'm good either way, having to exec into a container to run some dev tools doesn't really bother me, but I also don't think opening up a DB running in a container to whatever network environment your development setup is in (in most cases, probably your home router) is a huge security concern either.